pkg/distributors/contracts/PoolTokenCache.sol

```diff
@@ -30,13 +30,25 @@ contract PoolTokenCache {
     mapping(bytes32 => EnumerableSet.AddressSet) private _poolTokenSets;
     mapping(bytes32 => bool) private _poolTokenSetSaved;
 
+    function savePoolTokenSet(bytes32 poolId) public {
+        (IERC20[] memory poolTokens, , ) = vault.getPoolTokens(poolId);
+        if (_poolTokenSetSaved[poolId]) {
+            uint256 numTokens = _poolTokenSets[poolId].length();
+            for (uint256 t; t < numTokens; t++) {
+                // always the 0 index since we're removing all elements
+                address tokenAddress = _poolTokenSets[poolId].unchecked_at(0);
+                _poolTokenSets[poolId].remove(tokenAddress);
```

Comment:
This works, but it is inefficient for no good reason due a quirk on the OZ original implementation. Will follow up with an improvement.